src/Workspaces/Core/Portable/Diagnostics/DiagnosticAnalysisResult.cs

```diff
@@ -27,45 +27,27 @@ internal readonly struct DiagnosticAnalysisResult
     /// The set of documents that has any kind of diagnostics on it.
     /// </summary>
     public readonly ImmutableHashSet<DocumentId>? DocumentIds;
-    public readonly bool IsEmpty;
```

Comment:
redundant.

Reply:
also, on a personal note, i had data like this that needs to be passed in, as opposed to be computed and cached based on the data teh constructor is initailized with.  it's so easy to accidentally screw it up.